Sources/Commands/SwiftPackageCollectionsTool.swift

```diff
@@ -0,0 +1,358 @@
+        var jsonOptions: JSONOptions
+
+        @OptionGroup
+        var profileOptions: ProfileOptions
+
+        mutating func run() throws {
+            let collections = try await { self.collections.listCollections(identifiers: nil, in: profileOptions.usedProfile, callback: $0) }
+
+            if jsonOptions.json {
+                try JSONEncoder().print(collections)
+            } else {
+                collections.forEach {
+                    print("\($0.name) - \($0.source.url)")
+                }
+            }
+        }
+    }
+
+    struct Refresh: ParsableCommand {
+        static let configuration = CommandConfiguration(abstract: "Refresh configured collections")
+
+        @OptionGroup
+        var profileOptions: ProfileOptions
+
+        mutating func run() throws {
+            let collections = try await { self.collections.refreshCollections(in: profileOptions.usedProfile, callback: $0) }
+            print("Refreshed \(collections.count) configured package collections.")
+        }
+    }
+
+    struct Add: ParsableCommand {
+        static let configuration = CommandConfiguration(abstract: "Add a new collection")
+
+        @Argument(help: "URL of the collection to add")
+        var collectionUrl: String?
+
+        @Option(name: .long, help: "Sort order for the added collection")
+        var order: Int?
+
+        @OptionGroup
+        var profileOptions: ProfileOptions
+
+        mutating func run() throws {
+            guard let collectionUrlString = collectionUrl, let collectionUrl = URL(string: collectionUrlString) else {
+                throw CollectionsError.missingArgument("collectionUrl")
+            }
+
+            let source = PackageCollectionsModel.CollectionSource(url: collectionUrl)
+            let collection = try await { self.collections.addCollection(source, order: order, to: profileOptions.usedProfile, callback: $0) }
+
+            print("Added \"\(collection.name)\" to your package collections.")
+        }
+    }
+
+    struct Remove: ParsableCommand {
+        static let configuration = CommandConfiguration(abstract: "Remove a configured collection")
+
+        @OptionGroup
+        var profileOptions: ProfileOptions
+
+        @Argument(help: "URL of the collection to remove")
+        var collectionUrl: String?
+
+        mutating func run() throws {
+            guard let collectionUrlString = collectionUrl, let collectionUrl = URL(string: collectionUrlString) else {
+                throw CollectionsError.missingArgument("collectionUrl")
+            }
+
+            let collections = try await { self.collections.listCollections(identifiers: nil, in: profileOptions.usedProfile, callback: $0) }
+            let source = PackageCollectionsModel.CollectionSource(url: collectionUrl)
+
+            guard let collection = collections.first(where: { $0.source == source }) else {
+                throw CollectionsError.noCollectionMatchingURL(collectionUrlString)
+            }
```

Comment:
I am not sure we have to throw in this case, but then the extra `listCollections` is inexpensive so I am not against doing the extra check and letting user know of the error either.

Reply:
right now the business logic reports "happy" if the collection does not exist. this is typical for delete operations

Reply:
I think that makes sense from the API perspective, but in CLI tools, it's common to error in such cases, e.g.

```
$ rm does-not-exist
rm: does-not-exist: No such file or directory
```

Reply:
Agreed that silent success seems good for an API, but for CLI it might be good to at least have a warning that the key is already non-existent.